### PR TITLE
fix: Send ping to telemetry when search is engaged.

### PR DESCRIPTION
**Pull request by ericawright**
Fixes: #981

### CODE REVIEW

**Comment by ericawright** [comment]
I am also curious - but since enter/return never needs to be clicked we'd only be able to track keystrokes - I could look for the first keystroke?

**Comment by bwinton** [comment]
Yeah, or maybe also track how often we show the "n more tabs" button?

**Comment by ericawright** [comment]
True. Would you like me to change it?

**Comment by bwinton** [comment]
Sure.  I mean, you've got to add a semi-colon anyways…  :wink:

File: verticaltabs.js [hunk 1]
```diff
@@ -736,12 +736,22 @@ VerticalTabs.prototype = {
...
         this.clearFind();
       }
     });
+    find_input.onfocus = () => { this.sendPing('tab_center_search_focus', window);};
+    find_input.onblur = () => {
+      let details = {search_engaged: window.VerticalTabs.search_engaged};
+      this.sendPing('tab_center_search_blur', window, details);
+      window.VerticalTabs.search_engaged = false;
+    };
+    this.search_engaged = false;
```
Comment (bwinton) [inline verticaltabs.js]: Why are we sometimes calling `this.search_engaged` and other times `window.VerticalTabs.search_engaged`?  Shouldn't they be the same thing?

Comment (ericawright) [inline verticaltabs.js]: yup, not needed

File: utils.js [hunk 2]
```diff
@@ -90,7 +92,7 @@ function sendPing(key, window, details) {
   Object.assign(payload, details);
 
   let ping = JSON.stringify(payload);
-  // console.log('send ping: ' + ping); //debug: to check the ping details
+  console.log('send ping: ' + ping); //debug: to check the ping details
```
Comment (bwinton) [inline utils.js]: We probably want to leave this as a comment…

File: verticaltabs.js [hunk 3]
```diff
@@ -736,12 +736,22 @@ VerticalTabs.prototype = {
     search_icon.addEventListener('click', function (e) {
       find_input.focus();
     });
-    find_input.addEventListener('input', this.filtertabs.bind(this));
+    find_input.addEventListener('input', () => {
+      window.VerticalTabs.search_engaged = true;
```
Comment (bwinton) [inline verticaltabs.js]: Should this also be `this.search_engaged`, to match below?

File: utils.js (hunk 2 above)
Comment (ericawright) [inline utils.js]: oops

File: verticaltabs.js (hunk 3 above)
Comment (ericawright) [inline verticaltabs.js]: also, oops

**Pull request merged**